### PR TITLE
add some bfs & dfs problems.

### PR DESCRIPTION
**Pull request by wodeleeway**

### CODE REVIEW

File: python/bfs_topological_sort/lintcode_120_word_ladder.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        ans = float('inf')
+        visited = set([start])
+        
+        while queue:
+            curr = queue.popleft()
+            next_words = self.get_next_words(curr[0])
+            
+            for n in next_words:
+                if n == end:
+                    ans = min(ans, curr[1] + 1)
```
Comment (zjkang) [inline python/bfs_topological_sort/lintcode_120_word_ladder.py]: I think you should early return here once you find the end word?

File: python/bfs_topological_sort/lintcode_137_clone_graph.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                new_node.neighbors.append(new_neighbor)
+                
+                
+            
+                
+            
+        
+        
+        
+
```
Comment (zjkang) [inline python/bfs_topological_sort/lintcode_137_clone_graph.py]: remove extra blanks

File: python/dfs/lintcode_425_letter_combinations_of_a_phone_number.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        
+        return ans
+            
+            
+    def dfs(self, digits, index, formed, ans, mapping):
+        if index == len(digits):
+            ans.append(formed)
+            return
+        
+
```
Comment (zjkang) [inline python/dfs/lintcode_425_letter_combinations_of_a_phone_number.py]: remove extra line

File: python/bfs_topological_sort/lintcode_611_knight_shortest_path.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+                
+            
+                
+            
+        
+        
+        
+
+
```
Comment (zjkang) [inline python/bfs_topological_sort/lintcode_611_knight_shortest_path.py]: remove extra blanks

File: python/bfs_topological_sort/lintcode_137_clone_graph.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            curr_node = queue.popleft()
+            
+            for neighbor in curr_node.neighbors:
+                if neighbor not in visited:
+                    if neighbor in visited:
+                        continue
+                    visited.add(neighbor)
+                    queue.append(neighbor)
+        
+        return list(visited)
```
Comment (zjkang) [inline python/bfs_topological_sort/lintcode_137_clone_graph.py]: not necessary to convert to list?

File: python/bfs_topological_sort/lintcode_611_knight_shortest_path.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        
+        queue.append((source.x, source.y, 0))
+        
+        ans = float('inf')
+        
+        while queue:
+            x, y, count = queue.popleft()
+            
+            if x == destination.x and y == destination.y:
+                ans = min(ans, count)
```
Comment (zjkang) [inline python/bfs_topological_sort/lintcode_611_knight_shortest_path.py]: early return?

File: python/bfs_topological_sort/lintcode_120_word_ladder.py (hunk 1 above)
Comment (wodeleeway) [reply python/bfs_topological_sort/lintcode_120_word_ladder.py]: Addressed

File: python/bfs_topological_sort/lintcode_137_clone_graph.py (hunk 2 above)
Comment (wodeleeway) [reply python/bfs_topological_sort/lintcode_137_clone_graph.py]: Addressed

File: python/dfs/lintcode_425_letter_combinations_of_a_phone_number.py (hunk 3 above)
Comment (wodeleeway) [reply python/dfs/lintcode_425_letter_combinations_of_a_phone_number.py]: Addressed

File: python/bfs_topological_sort/lintcode_611_knight_shortest_path.py (hunk 4 above)
Comment (wodeleeway) [reply python/bfs_topological_sort/lintcode_611_knight_shortest_path.py]: Addressed

File: python/bfs_topological_sort/lintcode_137_clone_graph.py (hunk 5 above)
Comment (wodeleeway) [reply python/bfs_topological_sort/lintcode_137_clone_graph.py]: Addressed

File: python/bfs_topological_sort/lintcode_611_knight_shortest_path.py (hunk 6 above)
Comment (wodeleeway) [reply python/bfs_topological_sort/lintcode_611_knight_shortest_path.py]: Addressed

**Pull request merged**